### PR TITLE
[bug fix] DAGMC Advanced tallies were not producing vector tagged mes…

### PR DESCRIPTION
**Pull request by makeclean**
This PR modified the underlying routines to fix a number of issues regarding the DAGMC mesh tallies interfacing with pyne.r2s. The routines produced n_group+2 scalar tags, when we should have produced a single n_group sized vector tag. The +2 tags comes form 0-bottom group, and a total group tag. This has been modified for both the KDE tally and Tetmesh tracklength tally. 

A future fix could figure out how automatically tag the total group into a separate TOTAL_TAG, it may need some reworking of the logic, refactor into a common tagging function maybe?

### CODE REVIEW

**Comment by makeclean** [comment]
Dont merge yet, I want to add a unit test for this.

File: mcnp/meshtal_funcs.cpp
Comment (gonuke) [inline mcnp/meshtal_funcs.cpp]: Is this because `*n_energy_mesh` includes the bottom of the first bin (0.0 MeV) and thus had N+1 boundaries for N groups?

File: tally/TrackLengthMeshTally.cpp
Comment (gonuke) [inline tally/TrackLengthMeshTally.cpp]: But this version doesn't include the total at all... right?  Couldn't we just add an additional tag that is also the total?  In fact, you create the `total_tally_tag` above, but don't use it.

**Review by gonuke — CHANGES_REQUESTED** [review]
Don't we have all the pieces for including the total tally, but just fail to use it?

File: mcnp/meshtal_funcs.cpp
Comment (makeclean) [reply mcnp/meshtal_funcs.cpp]: Yes indeed

File: tally/TrackLengthMeshTally.cpp
Comment (makeclean) [reply tally/TrackLengthMeshTally.cpp]: So right now the Tally API forbids one from accessing tally group data with bins higher than N_GROUPS-1, we could easily change the API and then it would be a trivial change.

**Comment by makeclean** [comment]
Ok, I think this is good to go.

File: tally/CMakeLists.txt
Comment (ljacobson64) [inline tally/CMakeLists.txt]: What are these for?

Comment (makeclean) [reply tally/CMakeLists.txt]: This is the most cmakey way of enabling c++std's like c++11

Comment (ljacobson64) [reply tally/CMakeLists.txt]: Hmm ok. I don't see anything in here that looks like it needs c++11 though. I can build it fine without these 2 lines with gcc 4.4

Comment (ljacobson64) [reply tally/CMakeLists.txt]: Actually I lied, gcc-4.4 doesn't like std::to_string. I guess my main qualm here is that this PR would make it so it's no longer possible to build DAGMC on ACI with the default compiler

**Review by ljacobson64 — APPROVED** [review]
I changed my mind, I don't care about gcc 4.4 anymore.

**Comment by makeclean** [comment]
Just rebased this PR @cadangelo needs this for her work and so does a user here.

**Comment by ljacobson64** [comment]
CI seems to be passing for gcc-6 and gcc-7 but failing for gcc-5, clang-4.0, and clang-5.0. It doesn't seem to like the usage of `to_string`

**Comment by makeclean** [comment]
Yeah I've lost enable c++11 at some point, I'll fix and push again

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by cadangelo** [comment]
@makeclean I built this branch, ran neutron transport, then expanded the vector tag (TALLY_TAG).  There is one less group than there should be.  I haven't fully dug in, but I think the number of energy bins in TallyData.cpp is incorrect.

**Comment by makeclean** [comment]
@cadangelo could you send me associated inputs - or post them somewhere for me to download?

**Comment by cadangelo** [comment]
This PR is good to go!  My previous issue with the number of energy bins was my mistake.

**Comment by shimwell** [comment]
> This PR is good to go! My previous issue with the number of energy bins was my mistake.

Are the outstanding reviews for an outdated versions of this PR?

**Comment by ljacobson64** [comment]
@gonuke are you still looking to see changes here? Or this ok to merge?

**Comment by gonuke** [comment]
This is prob OK.  I was hoping to double check the treatment of the total flux, but that could be a different issue/PR if necessary - this seems to do what it’s supposed to

**Review by gonuke — APPROVED** [review]

**Comment by ljacobson64** [comment]
One last thing before I merge -- @makeclean can you add a file to the `news` directory explaining what this does?

**Comment by gonuke** [comment]
Please rebase against the latest updates so we can double check that this passes tests with the right versions of MOAB

File: src/tally/CMakeLists.txt
Comment (ljacobson64) [inline src/tally/CMakeLists.txt:9]: Don't need this anymore because we now have C++11 enabled by default

**Pull request closed without merging**
